docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,128 @@
+Except for providing the theme in the inner React context, this new provider has as a responsibility to generate CSS variables out of all tokens in the theme that are not functions, and make them available in the context.
+
+All these variables, are available under a key in the theme, named `vars`.
+The structure of this object is identical to the theme structure, the only difference is that the values represent some CSS varaibles.
+
+The best way to see this is by example.
+Let's say we have this element that uses some theme token:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivThemeToken.js", "hideToolbar": true }}
+
+If you try to inspect it, you will see that the color style has a direct hex value, that comes from some theme token:
+
+```html
+<div style="color: #2e7d32;">...</div>
+```
+
+Now, if the same div used CSS variables, as in the example below, you'll be able to track where that color token is defined in the theme:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.vars.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivCSSVariable.js", "hideToolbar": true }}
+
+```html
+<div style="color: var(--md-palette-success-main);">...</div>
+```
+
+Additionally, by clicking on the CSS variable in the browser's dev tools, you'll be taken to the theme object where you'll see its direct hex value again.
+We believe this to be a significant improvement in developer experience when debugging styles for any component.
+
+### Creating a custom theme
+
+The example above is using the default Material UI theme.
+But, what if you wanted to create your own custom theme?
+Let's see how to do that using the `experimental_extendTheme` utility.
+
+To start, it is important to understand this new theme structure.
+A theme is a collection of color schemes, each one defined within a single theme object.
+You'll be able to create more than light and dark color schemes if you want but, for the sake of simplicity, let's use only these two as an example. Here's how you'd customize it:
+
+```jsx
+const theme = experimental_extendTheme({
+  colorSchemes: {
+    light: {
+      palette: {
+        primary: teal,
+        secondary: deepOrange,
+      },
+    },
+    dark: {
+      palette: {
+        primary: cyan,
+        secondary: orange,
+      },
+    },
+  },
+});
+```
+
+{{"demo": "CssVarsCustomTheme.js", "iframe": true }}
+
+To toggle between color schemes, you can use the `useColorScheme` hook.
+This will make the browser re-write the CSS variables, pointing them to the new values you've created.
+The components' classes won't change at all, though—they should still refer to the same variables as before.
```

Comment:
"Should" here seems to imply that they _might_ not actually behave as expected.

```suggestion
This will make the browser re-write the CSS variables, pointing them to the new values you created—but the components' classes will still refer to the same variables as before.
```